pkg/util/workqueue/queue.go

```diff
@@ -113,6 +113,16 @@ func (q *Type) Get() (item interface{}, shutdown bool) {
 	return item, false
 }
 
+// Get the head of the queue
+func (q *Type) Head() interface{} {
+	// We do not need lock here
```

Comment:
personally, I think reading usually does not require a lock. Actually, you are right, it is better to have a lock. I will update the code

Thanks